xxxApp/src/Makefile

```diff
@@ -394,15 +381,14 @@ ifneq (,$(findstring linux, $(T_A)))
   endif
 endif
 
-$(PROD_NAME)_LIBS := $($(PROD_NAME)_LIBS) $(EPICS_BASE_IOC_LIBS)
+$(PROD_NAME)_LIBS += $(EPICS_BASE_IOC_LIBS)
 
 # <name>_registerRecordDeviceDriver.cpp will be created from <name>.dbd
-$(PROD_NAME)_SRCS_vxWorks  += $(DBD_NAME)_registerRecordDeviceDriver.cpp
-$(PROD_NAME)_SRCS_Linux    += $(DBD_NAME)_registerRecordDeviceDriver.cpp    $(PROD_NAME)Main.cpp
-$(PROD_NAME)_SRCS_WIN32    += $(DBD_NAME)_registerRecordDeviceDriver.cpp    $(PROD_NAME)Main.cpp
-$(PROD_NAME)_SRCS_cygwin32 += $(DBD_NAME)_registerRecordDeviceDriver.cpp    $(PROD_NAME)Main.cpp
-$(PROD_NAME)_SRCS_solaris  += $(DBD_NAME)_registerRecordDeviceDriver.cpp    $(PROD_NAME)Main.cpp
-$(PROD_NAME)_SRCS_Darwin   += $(DBD_NAME)_registerRecordDeviceDriver.cpp    $(PROD_NAME)Main.cpp
+ifeq ($(OS_CLASS), vxWorks) 
+  $(PROD_NAME)_SRCS  += $(DBD_NAME)_registerRecordDeviceDriver.cpp
+else
+  $(PROD_NAME)_SRCS  += $(DBD_NAME)_registerRecordDeviceDriver.cpp $(PROD_NAME)Main.cpp
```

Comment:
Looks good to me.

Here's a trivial refinement that avoids repeating the RRDD.cpp source name.
```make
$(PROD_NAME)_SRCS += $(DBD_NAME)_registerRecordDeviceDriver.cpp

ifneq ($(OS_CLASS), vxWorks)
  $(PROD_NAME)_SRCS += $(PROD_NAME)Main.cpp
endif
```
or maybe even replace that last conditional with
```make
$(PROD_NAME)_SRCS_DEFAULT = $(PROD_NAME)Main.cpp
$(PROD_NAME)_SRCS_vxWorks = -nil-
```